test/test-helpers/did-render.ts

```diff
@@ -0,0 +1,13 @@
+import Application from '../../src/application';
+
+async function didRender(app: Application): Promise<void> {
+  return new Promise<void>(resolve => {
+    let watcher = setInterval(function() {
+      if (app['_scheduled']) return;
```

Comment:
I don't think we can change the order of setting _scheduled (if we did then we could never trigger a rerender while rerendering). We can set another flag though that is essentially set at the same time as _scheduled and is unset after all rendering is completed.